### ISSUE TITLE
Separation of properties does not work very well

### ISSUE THREAD

**Issue by GolovanovSrg** [ananyahjha93/cycle-consistent-vae#1]
I use the code of your repository with standard parameters. After learning the autoencoder on mnist, I train the classifier (from your repository). When using class_latent_space over 100 epochs, I get the accuracy of 0.98 on the test. When using style_mu for the same 100 epochs on the test - 0.88.

**Comment by ananyahjha93** [comment]
Hey, I have not included any code for a classifier in the repositories. Also, this happens with MNIST when the learning rate is not lowered according to the scheduler provided in the training.py file. The uploaded models in the checkpoints folder contain models produced using the very same training code uploaded here. You can verify the results using those saved files.

**Comment by GolovanovSrg** [comment]
To learn the classifier I use the following code (the architecture of the classifier from your repository):

```
def classifier_training_procedure(FLAGS):
    encoder = Encoder(style_dim=FLAGS.style_dim, class_dim=FLAGS.class_dim)
    encoder.apply(weights_init)

    if FLAGS.load_saved:
        encoder.load_state_dict(torch.load(os.path.join('checkpoints', FLAGS.encoder_save)))

    classifier = Classifier(z_dim=FLAGS.class_dim, num_classes=10)

    optimizer = optim.Adam(classifier.parameters(), lr=FLAGS.initial_learning_rate)
    criterion = nn.CrossEntropyLoss()

    if FLAGS.cuda:
        encoder = encoder.cuda()
        classifier = classifier.cuda()

    train_mnist = datasets.MNIST(root='mnist', download=True, train=True, transform=transform_config)
    train_loader = DataLoader(train_mnist, batch_size=FLAGS.batch_size, shuffle=True, num_workers=0)

    test_mnist = datasets.MNIST(root='mnist', download=True, train=False, transform=transform_config)
    test_loader = DataLoader(test_mnist, batch_size=FLAGS.batch_size, shuffle=False, num_workers=0)

    encoder.eval()
    for epoch in range(FLAGS.start_epoch, FLAGS.end_epoch):
        classifier.train()
        tqdm_train_data = tqdm(train_loader, desc='Train (epoch #{})'.format(epoch))
        for batch, label in tqdm_train_data:
            if FLAGS.cuda:
                batch, label = batch.cuda(), label.cuda()

            style_mu, style_logvar, _ = encoder(batch)
            outputs = classifier(style_mu)
            loss = criterion(outputs, label)

            optimizer.zero_grad()
            loss.backward()
            optimizer.step()

            tqdm_train_data.set_postfix({'loss': loss.item()})

        classifier.eval()
        tqdm_test_data = tqdm(test_loader, desc='Test (epoch #{})'.format(epoch))
        acc, acc_total = 0, 0
        for batch, label in tqdm_test_data:
            if FLAGS.cuda:
                batch, label = batch.cuda(), label.cuda()

            style_mu, _, _ = encoder(batch)
            outputs = classifier(style_mu)
            loss = criterion(outputs, label)

            predicted = outputs.argmax(dim=-1)
            acc += torch.sum(predicted == label).item()
            acc_total += batch.shape[0]

            tqdm_test_data.set_postfix({'loss': loss.item(), 'accuracy': acc / acc_total})
```

I also tried to use weights from the checkpoints folder from the repository. When loading weights there were problems, because in the saved files, the layers of the encoder and the decoder were called differently than in the code. Fortunately, this problem could easily be remedied.

As a result, with style_mu and provided weights I still get a good classification accuracy.

**Comment by ananyahjha93** [comment]
Hey, first of all, thank you for pointing out, I did include the architecture for the classifier in the networks file. OK, the reason you are getting higher classification accuracy than claimed in the paper is because you are directly classifying style_mu. In a VAE, style_mu is just one of the parameters of the standard normal that we predict. This part is mentioned in the paper and even used in the works of Mathieu et al. (against which we have compared our results); for classification we do not just use style_mu. Rather, we take both the mu and logvar, sample from that distribution and then use that point in the classifier. As mentioned, this is one of the major factors for adding class confusion in the style latent space, something which the non-VAE architecture of Szabo et al. lacks and hence has such high classification accuracies.
